src/Spawner.cpp

```diff
@@ -186,40 +186,47 @@ bool Spawner::ProcessHandle::alive() const
 
     int status = 0;
     pid_t ret = waitpid(pid, &status, WNOHANG);
-    
-    if(ret < 0 )
+    //waitpid(): on success, returns the process ID of the child whose state has
+    //changed; if WNOHANG was specified and one or more child(ren) specified by
+    //pid exist, but have not yet changed state, then 0 is returned.
+    //On error, -1 is returned.
+    if(ret == -1 )
     {
         throw std::runtime_error(std::string("WaitPid failed ") + strerror(errno));
     }
-    
-    if(!status)
-    {
+    else if(ret == 0){
+        //Not yet terminated, but also no error.. so it's still alive
+        std::cout << "Process " << pid << " is still running" << std::endl;
```

Comment:
I'm not sure how often `alive()` is called, but this might be better as a debug output.

Reply:
It's possible that it gets called a few times before the process was terminated. If we had here a proper text-logging (eg base-logging) already in this package, I'd chosen debug print... so I agree with your comment.

But I'd prefer to consider integrating a text logging library as a separated issue. And in that go a categorization of all textlogs we have here in orocos_cpp (see #24).  @saarnold, please resolve conversation if you agree